error.go

```diff
@@ -2,38 +2,42 @@
 // and efficient error handling.
 package microerror
 
-import (
-	"fmt"
-
-	"github.com/juju/errgo"
-)
-
 var (
-	// MaskAny is a simple error masker. Masked errors act as tracers within the
-	// source code. Inspecting an masked error shows where the error was passed
-	// through within the code base. This is gold for debugging and bug huntin.
-	MaskAny = errgo.MaskFunc(errgo.Any)
+	Default = NewErrgoHandler(ErrgoHandlerConfig{
+		CallDepth: 1,
```

Comment:
I checked the last commit and it works as expected. The question is if you want to users to use the default. You are quite educated and think certain scenarios do not happen, but they just happen. This is why we should try to prevent making mistakes by design. When you want `Default` to be used by clients just add tests. When you do not want this make it private.